hw2/stackCalculator/StackOnArray.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+
+namespace StackCalculator
+{
+    public class StackOnArray : IStack
+    {
+        private double[] stack;
+        private const int stackSize = 10;
+        private int count;
+        public StackOnArray()
+        {
+            stack = new double[stackSize];
+        }
+        public StackOnArray(int size)
+        {
+            stack = new double[size];
+        }
+        // Добавляет элемент
+        public void Push(double element)
+        {
+            if (count == stack.Length)
+            {
+                Array.Resize(ref stack, stack.Length + 10);
+            }
+            stack[count++] = element;
+        }
+        // Возвращает, пустой ли стек
+        public bool IsEmpty
+        {
+            get { return count == 0; }
+        }
```

Comment:
Вот только этой штуки в интерфейсе нет, поэтому ей почти никак не воспользоваться. И можно короче записать:
```suggestion
        public bool IsEmpty => count == 0;
```